<a href="https://colab.research.google.com/github/Gabriel-Orlandi-Portes/Analise-Album-Copa2026/blob/main/analise_album_copa2026.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1. 📊 Análise das Simulações do Álbum da Copa do Mundo 2026

Este notebook tem como objetivo analisar os dados gerados através das simulações do álbum da Copa do Mundo 2026.

A simulação foi criada em Python e tem como objetivo estimar:

- Quantidade de pacotes necessários para completar o álbum
- Custo total da coleção
- Quantidade de figurinhas repetidas
- Diferença entre repetidas normais e brilhantes
- Distribuição dos resultados obtidos

Os dados utilizados neste notebook foram gerados através do arquivo `1000_simulacoes.py`.




---


# 2. 📚 Importação das Bibliotecas

Nesta etapa serão importadas as bibliotecas utilizadas durante a análise:

- Pandas: manipulação e análise dos dados
- Matplotlib: criação das visualizações gráficas

In [2]:
import pandas as pd
import matplotlib.pyplot as plt

print('Bibliotecas importadas com sucesso!')

Bibliotecas importadas com sucesso!




---


# 3. 📥 Carregamento e Leitura dos Dados

Os dados utilizados na análise estão armazenados no arquivo `simulacoes.csv`.

*   O arquivo será carregado utilizando o Pandas para criação de um DataFrame.

*   O DataFrame será utilizado para realizar todas as análises posteriores.

In [3]:
# Selecionar arquivo simulacoes.csv

from google.colab import files

uploaded = files.upload()

Saving simulacoes.csv to simulacoes.csv


In [4]:
# Carregando a tabela...

df = pd.read_csv('simulacoes.csv')

print('Tabela carregada com sucesso!')

Tabela carregada com sucesso!


---


# 4. 🔎 Exploração Inicial dos Dados

Antes de iniciar as análises, será realizada uma exploração inicial do dataset.

O objetivo é entender:

- Estrutura dos dados
- Quantidade de registros
- Tipos das variáveis
- Estatísticas básicas

In [5]:
# Visualização inicial dos dados.

df.head()

,pacotes,custo,repetidas_total,repetidas_normais,repetidas_brilhantes
0,1160,8195,7140,6650,490
1,944,6683,5628,5187,441
2,833,5906,4851,4486,365
3,854,6053,4998,4655,343
4,1055,7460,6405,5940,465


In [ ]:
# Quantidade de linhas e colunas.

df.shape

(1000, 5)

In [ ]:
# Variáveis e dados

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 5 columns):
 #   Column                Non-Null Count  Dtype
---  ------                --------------  -----
 0   pacotes               1000 non-null   int64
 1   custo                 1000 non-null   int64
 2   repetidas_total       1000 non-null   int64
 3   repetidas_normais     1000 non-null   int64
 4   repetidas_brilhantes  1000 non-null   int64
dtypes: int64(5)
memory usage: 39.2 KB


In [ ]:
# Dados Estatísticos da tabela

df.describe()

,pacotes,custo,repetidas_total,repetidas_normais,repetidas_brilhantes
count,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000
mean,1045.560000,7393.920000,6338.920000,5898.275000,440.645000
std,182.725595,1279.079165,1279.079165,1190.468144,91.409292
min,669.000000,4758.000000,3703.000000,3412.000000,266.000000
25%,920.000000,6515.000000,5460.000000,5074.750000,376.000000
50%,1013.000000,7166.000000,6111.000000,5684.500000,426.000000
75%,1131.250000,7993.750000,6938.750000,6457.000000,486.000000
max,1881.000000,13242.000000,12187.000000,11298.000000,889.000000


In [6]:
# Padronizando nome das colunas para facilitar a análise

df.rename(
    columns={
        "pacotes": "pacotes_comprados",
        "custo": "custo_total"
    },
    inplace=True
)



---


#5. 📦 Análise da Quantidade de Pacotes Necessários

Serão analisadas medidas estatísticas para compreender a distribuição dos resultados:

- Média
- Mediana
- Quartis
- Valores mínimos e máximos

Essas métricas permitem avaliar não apenas o resultado médio, mas também a variação entre diferentes simulações.

### **5.1 Média**

A média é uma medida de tendência central que representa o valor médio de um conjunto de dados.

Ela é calculada através da soma de todos os valores observados, dividida pela quantidade total de elementos.

Neste projeto, a média representa a quantidade esperada de pacotes necessários para completar o álbum considerando as 1.000 simulações realizadas.

O cálculo foi realizado utilizando a coluna `pacotes_comprados`. A função **mean()** da biblioteca Pandas realiza automaticamente o cálculo da média.

In [11]:
media_pacotes = df["pacotes_comprados"].mean()

print(f"Média de pacotes necessários: {media_pacotes:.0f}")

Média de pacotes necessários: 1046


### **Conclusão**

Com base nas 1.000 simulações realizadas, foi encontrada uma média de aproximadamente **1046 pacotes necessários** para completar o álbum.

Isso significa que, seguindo o modelo da simulação e sem realizar trocas de figurinhas, um colecionador precisaria comprar em torno desse valor de pacotes para conseguir todas as figurinhas.

---

### **5.2 Valores Mínimos e Máximos**

Os valores mínimo e máximo representam os extremos encontrados dentro das simulações.

O valor mínimo indica a menor quantidade de pacotes necessária para completar o álbum entre todos os testes realizados.

Já o valor máximo representa o cenário onde foi necessário comprar a maior quantidade de pacotes.

Essas medidas ajudam a observar a variação existente entre diferentes simulações.

Também utiliza funções da biblioteca Pandas **(min() e max())**

In [12]:
menor_quantidade = df["pacotes_comprados"].min()
maior_quantidade = df["pacotes_comprados"].max()

print(f"Menor quantidade encontrada: {menor_quantidade}")
print(f"Maior quantidade encontrada: {maior_quantidade}")

Menor quantidade encontrada: 669
Maior quantidade encontrada: 1881


### **Conclusão**

Entre todas as simulações realizadas, o melhor cenário conseguiu completar o álbum com apenas **669 pacotes**, enquanto o pior cenário precisou de **1881 pacotes**.

Essa diferença mostra o impacto da aleatoriedade na compra dos pacotes, já que cada simulação possui resultados diferentes mesmo utilizando as mesmas regras.

---

### **5.3 Mediana**

A mediana representa o valor central de um conjunto de dados ordenados.

Isso significa que metade das simulações apresentou um resultado menor ou igual à mediana, enquanto a outra metade apresentou um resultado maior ou igual.

Essa métrica é importante porque reduz o impacto de valores muito altos ou muito baixos que podem influenciar a média.

In [8]:
mediana_pacotes = df["pacotes_comprados"].median()

print(f"Mediana de pacotes necessários: {mediana_pacotes:.0f}")

Mediana de pacotes necessários: 1013


### **Conclusão**

A mediana encontrada foi de **1013 pacotes**.

Isso significa que metade das simulações conseguiu completar o álbum utilizando uma quantidade menor ou igual a esse valor, enquanto a outra metade precisou de uma quantidade maior.

Como a mediana ficou próxima da média de **1046 pacotes**, os resultados apresentam uma distribuição equilibrada, sem grande influência de valores muito fora do padrão.

---

### **5.4 Quartis**

Os quartis dividem os dados em quatro partes iguais, permitindo analisar melhor a distribuição dos resultados.

Eles ajudam a identificar como os valores estão concentrados dentro das simulações.

Neste projeto serão analisados:

- Primeiro quartil (25%)
- Mediana (50%)
- Terceiro quartil (75%)

Essas informações mostram a faixa onde está concentrada a maior parte dos resultados.

In [9]:
quartis_pacotes = df["pacotes_comprados"].quantile([0.25, 0.50, 0.75])

quartis_pacotes

,pacotes_comprados
0.25,920.00
0.50,1013.00
0.75,1131.25


## **Conclusão**

Os quartis permitem observar onde está concentrada a maior parte das simulações.

O intervalo entre o primeiro e o terceiro quartil representa a faixa onde estão 50% dos resultados centrais.

Dessa forma, é possível entender não apenas o valor médio esperado, mas também a variação normal dos resultados obtidos.